### DISCUSSION TITLE
Add a new record

### DISCUSSION THREAD

**Discussion by BerengereS** [dotenx/dotenx#219] [Show and tell]
Can you explain to me step by step how to add a new record in a table from the fields entered in a form.
Thanks !

**Comment by mkamrani** [comment]
Hey, sure. I can create a demo video for you. Just need some more details.

Are you using DoTenX UI-builder or you're using your custom code (say in React or something else)?

**Reply by BerengereS** [reply]
Hello, yes I use the UI Builder and I create my databases directly on DoTenX.
Thanks !

**Reply by mkamrani** [reply]
Sorry for the delay, actually found a bug I had to fix too.

Let's say you have a table with these fields:

name, age, location

Now in the UI builder do the followings:

1. Create a form
2. Add three inputs to the form and set the `name` of each input exactly the same as your table fields
3. Add a submit button
4. Copy the `Add record` endpoint of the table
![Screen Shot 2023-01-16 at 10 50 54 am](https://user-images.githubusercontent.com/15846333/212574262-322b9105-1f5b-4f2b-bc33-e087e28e3ec8.png)
5. Select your form and in the options tap (on the right panel), add the request handler and paste the url you copied in the URL section. Make sure you set the method to `POST`.
![Screen Shot 2023-01-16 at 10 51 18 am](https://user-images.githubusercontent.com/15846333/212574276-0fa984de-e68c-4398-96ae-b45bdc753109.png)

That's it.

**Comment by BerengereS** [comment]
Thank you for that answer. 

However I'm sorry but I think I missed something. I performed the steps described but nothing happened. I have trouble understanding in particular the link between the form and clicking on the submit button.

I am wondering whether to do an event on the submit button and also the difference between the submit and onclick event of the submit button.

Thanks
